crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -1726,7 +1754,17 @@ impl<'genv, 'tcx: 'genv, P: ConvPhase<'genv, 'tcx>> ConvCtxt<P> {
             fhir::ExprKind::Var(var, _) => {
                 match var.res {
                     ExprRes::Param(..) => env.lookup(var).to_expr(),
-                    ExprRes::Const(def_id) => rty::Expr::const_def_id(def_id).at(espan),
+                    ExprRes::Const(def_id) => {
+                        if P::HAS_ELABORATED_INFORMATION {
+                            let info = self.genv().constant_info(def_id)?;
+                            rty::Expr::const_def_id(def_id, info).at(espan)
```

Comment:
I'm not 100% sure, because both behaviors make sense and it's mostly about defaults, but  maybe we should report an error here for uninterpreted consts, e.g., in this case:


```rust
#[opaque]
#[refined_by(x: int)]
struct S { ... }

const SOME_S: S = ...

#[sig(fn() -> S[SOME_S])]
fn foo() -> S { ... }
```

```
#[sig(fn() -> S[SOME_S])]
                ^^^^^^
                sorry we can't deal with this constant
```

More generally, `ConstInfo::Uninterpreted` could be interpreted (pun intended) in two different ways

1. Sorry we can't deal with this so don't bother trying
2. Encode this constant with an uninterpreted (opaque) constant in smt

I think 1. by default is less surprising (or more discoverable) and the user should opt-in to the second behavior by annotating the const. You can pick 2. to circumvent these limitations _or_ to hide implementation details even if we support the constant.

Note that for 2. to make sense, we need to give `SOME_S` (when used in the mir) type `SOME_S[SOME_S]` and the behavior right now is the worst of both worlds because we encode `SOME_S` as an uninterpreted constant in smt but we don't give the rust const a useful type (we give it `∃v. SOME_S[v]`)